bcda/api/requests.go

```diff
@@ -173,21 +179,24 @@ func (h *Handler) bulkRequest(resourceTypes []string, w http.ResponseWriter, r *
 	// their job finishes or time expires (+24 hours default) for any remaining jobs left in a pending or in-progress state.
 	// Overall, this will prevent a queue of concurrent calls from slowing up our system.
 	// NOTE: this logic is relevant to PROD only; simultaneous requests in our lower environments is acceptable (i.e., shared opensbx creds)
-	if (os.Getenv("DEPLOYMENT_TARGET") == "prod") &&
-		(!db.Find(&pendingAndInProgressJobs, "aco_id = ? AND status IN (?, ?)", acoID, "In Progress", "Pending").RecordNotFound()) {
-		if types, err := check429(pendingAndInProgressJobs, resourceTypes, version); err != nil {
-			if _, ok := err.(duplicateTypeError); ok {
-				w.Header().Set("Retry-After", strconv.Itoa(utils.GetEnvInt("CLIENT_RETRY_AFTER_IN_SECONDS", 0)))
-				w.WriteHeader(http.StatusTooManyRequests)
+	if os.Getenv("DEPLOYMENT_TARGET") == "prod" {
+		err := db.First(&pendingAndInProgressJobs, "aco_id = ? AND status IN (?, ?)", acoID, "In Progress", "Pending").Error
+
+		if !goerrors.Is(err, gorm.ErrRecordNotFound) {
+			if types, err := check429(pendingAndInProgressJobs, resourceTypes, version); err != nil {
+				if _, ok := err.(duplicateTypeError); ok {
+					w.Header().Set("Retry-After", strconv.Itoa(utils.GetEnvInt("CLIENT_RETRY_AFTER_IN_SECONDS", 0)))
+					w.WriteHeader(http.StatusTooManyRequests)
+				} else {
+					log.Error(err)
+					oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.Processing, "")
+					responseutils.WriteError(oo, w, http.StatusInternalServerError)
+				}
+
+				return
 			} else {
-				log.Error(err)
-				oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.Processing, "")
-				responseutils.WriteError(oo, w, http.StatusInternalServerError)
+				resourceTypes = types
 			}
-
-			return
-		} else {
-			resourceTypes = types
 		}
```

Comment:
I can't comment on this unfortunately, but in the JobStatus function, we are using db.Find(), I believe we're relying on the ErrRecordNotFound, which I think requires us to use db.First()